### DISCUSSION TITLE
How to automatically reload configuration?

### DISCUSSION THREAD

**Discussion by alienman5k** [ghostty-org/ghostty#3643] [Q&A] [answered]
Is there a way to automatically reload configuration on saving config file?

**Comment by jcollie** [comment]
Not within Ghostty.

**Comment by mitchellh** [comment, marked as answer]
This was discussed some time back and we're not currently interested in adding this.

**Reply by alienman5k** [reply]
Thanks for the quick reply, any link to the the previous discussion?

**Reply by dusktreader** [reply]
This is pretty fundamental behavior when you are working on configuring an editor. Very disappointing answer.

**Reply by Davincible** [reply]
I'd have to agree, relatively basic feature built into most similar programs. Any specific reason you decided against it?

**Comment by syhol** [comment]
Is there a way I can DIY this? Can I reload Ghostty config from an external process? If not would you be open to adding `ghostty +reload-config` command?

**Reply by ernstwi** [reply]
One option is to use AppleScript:

```
tell application "System Events"
    tell process "Ghostty"
        click menu item "Reload Configuration" of menu "Ghostty" of menu bar item "Ghostty" of menu bar 1
    end tell
end tell
```

You can save this as `reload.scpt` and trigger with `osascript reload.scpt`

**Reply by raylee** [reply]
Thanks @ernstwi! I used this plus the [Simple Hotkey Daemon](https://github.com/koekeishiya/skhd) to wire up hotkeys to flip themes (light/dark) and adjust opacity. In case it's useful to the OP, the code is here: https://gist.github.com/raylee/283c0299e32926334f3ff5bf7f3e280d , but only applicable for MacOS at the moment.

This was one of the last things I was missing from WezTerm, which allows Lua code to be executed on hooks to build these sorts of things internal to the app.

**Comment by rossvz** [comment]
I'm using hyprland + ghostty and wanted a way to trigger this programmatically.
This is a bit of a hack but it works!

This assumes hyprland and that you haven't manually overwritten the keyboard shortcut to reload the config.

```bash
# Trigger ghostty config reload
if pgrep -x ghostty &> /dev/null; then
  ghostty_addresses=$(hyprctl clients -j | jq -r '.[] | select(.class == "com.mitchellh.ghostty") | .address')
  
  if [[ -n "$ghostty_addresses" ]]; then
    # Save current active window
    current_window=$(hyprctl activewindow -j | jq -r '.address')
    
    # Focus each ghostty window and send reload key combo
    while IFS= read -r address; do
      hyprctl dispatch focuswindow "address:$address"
      sleep 0.1
      hyprctl dispatch sendshortcut "CTRL SHIFT, comma, address:$address"
    done <<< "$ghostty_addresses"
    
    # Return focus to original window
    if [[ -n "$current_window" ]]; then
      hyprctl dispatch focuswindow "address:$current_window"
    fi
  fi
fi
```

**Reply by thebrahman** [reply]
this works great. now I have ghostty live reloading themes in omarchy!

**Reply by rossvz** [reply]
By the way this will be built-in to Omarchy when the next version of Ghostty is released. We have an open PR to connect ghostty theme for each of the default Omarchy themes and it will reload with the SIGUSR2 signal automatically.

**Reply by rossvz** [reply]
https://github.com/basecamp/omarchy/pull/358

**Comment by 00-kat** [comment]
By the way, since #7759 (which is only available on tip and [will be present in the next release][milestone]), you can get Ghostty to reload the configuration by sending it `SIGUSR2`, so you can run something like this in the background:
```sh
find ~/Library/Application\ Support/com.mitchellh.ghostty ~/.config/ghostty 2>/dev/null | entr pkill -SIGUSR2 ghostty
```
(The first path is only relevant on macOS, and this uses [`entr`] as an example.)

[milestone]: https://github.com/ghostty-org/ghostty/pull/7759#event-18420716057
[`entr`]: http://eradman.com/entrproject

**Comment by nicolaygerold** [comment]
osascript -e 'tell application "System Events" to tell process "ghostty" to keystroke "," using {command down, shift down}'

I am using this. Fires off the shortcut. Hacky but works.

**Reply by HelloThisIsFlo** [reply]
Thank you so much 🙏

**Comment by trevin-j** [comment]
Did some digging. Using the command `systemctl reload --user app-com.mitchellh.ghostty.service` you can remotely tell ghostty to reload. So, you can turn this into a watcher script to reload on file changes like below:

```bash
#!/usr/bin/env bash

CONFIG="$HOME/.config/ghostty/config"
SERVICE="app-com.mitchellh.ghostty.service"

while inotifywait -e close_write,move,create,delete "$CONFIG" >/dev/null 2>&1; do
  systemctl reload --user "$SERVICE"
done
```

This is distro/wm/de agnostic I think so it should work with whatever linux setup you have. Not Mac tho. But I bet there is a way to wire this script to execute whenever you run ghostty so it automatically works without you having to manually call it from hyprland.conf or some other startup config.

**Reply by rhodes-b** [reply]
https://github.com/ghostty-org/ghostty/blob/main/dist/linux/systemd.service.in#L9

like noted its just sending SIGUSR2 signal to the running ghostty instance

so a much more cross platform way would be `kill -s USR2 $(pgrep ghostty)` I assume `Ghostty.app` on macos im not sure
or use `pkill` like kat noted just above this https://github.com/ghostty-org/ghostty/discussions/3643#discussioncomment-13899379
